server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -0,0 +1,385 @@
+            if ((upperTermHasDecimalPart == false && includeHigh == false) || (upperTermHasDecimalPart && signum(parsedHigh) < 0)) {
+                if (parsedHigh.equals(defaultMinimum())) {
+                    return new MatchNoneFilter();
+                }
+                --parsedHigh;
+            }
+            return new RangeMatchDimFilter(mappedFieldType.name(), parsedLow, parsedHigh, true, true);
+        }
+        return null;
+    }
+
+    abstract Long defaultMinimum();
+
+    abstract Long defaultMaximum();
+
+}
+
+class IntegerFieldMapperNumeric extends NumericNonDecimalMapper {
+    @Override
+    Long defaultMinimum() {
+        return (long) Integer.MIN_VALUE;
+    }
+
+    @Override
+    Long defaultMaximum() {
+        return (long) Integer.MAX_VALUE;
+    }
+}
+
+class SignedLongFieldMapperNumeric extends NumericNonDecimalMapper {
+    @Override
+    Long defaultMinimum() {
+        return Long.MIN_VALUE;
+    }
+
+    @Override
+    Long defaultMaximum() {
+        return Long.MAX_VALUE;
+    }
+}
+
+abstract class NumericDecimalFieldMapper extends NumericMapper {
+
+    @Override
+    public DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues) {
+        if (mappedFieldType instanceof NumberFieldType) {
+            NumberFieldType numberFieldType = (NumberFieldType) mappedFieldType;
+            List<Object> convertedValues = new ArrayList<>(rawValues.size());
+            for (Object rawValue : rawValues) {
+                convertedValues.add(convertToDocValues(numberFieldType.numberType().parse(rawValue, true)));
+            }
+            return new ExactMatchDimFilter(mappedFieldType.name(), convertedValues);
+        }
+        return null;
+    }
+
+    @Override
+    public DimensionFilter getRangeMatchFilter(
+        MappedFieldType mappedFieldType,
+        Object rawLow,
+        Object rawHigh,
+        boolean includeLow,
+        boolean includeHigh
+    ) {
+        if (mappedFieldType instanceof NumberFieldType) {
+            NumberFieldType numberFieldType = (NumberFieldType) mappedFieldType;
+            Number l = Long.MIN_VALUE;
+            Number u = Long.MAX_VALUE;
+            if (rawLow != null) {
+                l = numberFieldType.numberType().parse(rawLow, false);
+                if (includeLow == false) {
+                    l = getNextHigh(l);
+                }
+                l = convertToDocValues(l);
+            }
+            if (rawHigh != null) {
+                u = numberFieldType.numberType().parse(rawHigh, false);
+                if (includeHigh == false) {
+                    u = getNextLow(u);
+                }
+                u = convertToDocValues(u);
+            }
+            return new RangeMatchDimFilter(numberFieldType.name(), l, u, true, true);
+        }
+        return null;
+    }
+
+    abstract long convertToDocValues(Number parsedValue);
```

Comment:
nit : can we rename this to something better - convertToIndexedValue or something similar